### PR TITLE
Fixes #251

### LINKED ISSUE #251: allow_unknown inside *of rule is ignored
When trying to use `allow_unknown` within an `*of` clause, validation fails. 
This is perhaps related to #250 , but as no exception is thrown in this case and the error tree is seemingly successfully built and only ignores the `allow_unknown` flag, I'm opening a separate issue.

```python

# This is the schema that causes the problem
schema1 = {
    'test': {
        'oneof': [
            {
                'type': 'dict', 
                'allow_unknown':True, 
                'schema': {'a': {'type': 'string'}}
            }
        ]
    }
}

# This is a control schema, meaning it is the same as the above, 
# just without wrapping the `oneof` clause around the inner `'dict'`
schema2 = {
    'test': {
        'type': 'dict', 
        'allow_unknown':True, 
        'schema': {'a': {'type': 'string'}}
    }
}

v1 = cerberus.Validator(schema1) 
v1.validate({'test': {'a': 's'}}) #  returns True (as expected) 
v1.validate({'test': {'a': 's', 'b': 'asd'}}) #  returns False (is `allow_unknown` ignored?)
print(v1.errors) #  {'test': [{'oneof': ['none or more than one rule validate', {'oneof definition 0': ['unknown field']}]}]}

v2 = cerberus.Validator(schema2)  
v2.validate({'test': {'a': 's', 'b': 'asd'}}) #  returns True (as expected)
```

### PR DESCRIPTION
**Pull request by daviskirk**
- add test

### CODE REVIEW

**Comment by daviskirk** [comment]
As this wasn't really related to #250 after all I looked at this as well while I was at it. As far as I can see, this is now the intended behaviour. 

I added a test for oneof, assuming the other *of rules should work the same.  Otherwise I can add the others too.... it might be alot of duplicated code that might just result in harder to maintain tests (As a side note I saw #213 and pytest parametrized tests are great cutting down on that sort of stuff).

File: cerberus/tests/tests.py [hunk 1]
```diff
@@ -1580,6 +1580,36 @@ def test_coerce_non_digit_in_sequence(self):
...
+                    },
+                ]
+            }
+        }
+        # check regression and that allow unknown does not cause any different
+        # than expected behaviour for one-of.
+        document = {'test': {'known': 's'}}
+        self.validator(document, schema)
+        _errors = self.validator._errors
+        self.assertEqual(len(_errors), 1)
```
Comment (funkyfuture) [inline cerberus/tests/tests.py]: please also check that there are no child-errors.

File: cerberus/tests/tests.py [hunk 2]
```diff
@@ -1580,6 +1580,36 @@ def test_coerce_non_digit_in_sequence(self):
...
+        self.assertError('test', ('test', 'oneof'),
+                         errors.ONEOF, schema['test']['oneof'],
+                         v_errors=_errors)
+        # check that allow_unknown is actually applied
```
Comment (funkyfuture) [inline cerberus/tests/tests.py]: *of-rules are ignored during normalization. am i missing something? otherwise i see no point in it.

Comment (daviskirk) [inline cerberus/tests/tests.py]: Yes, you're 100% correct... testing normalization makes no sense.

**Comment by nicolaiarocci** [comment]
Merged after rebase: https://github.com/nicolaiarocci/cerberus/commit/78704de458d50db0f38fb614aef1eeb09c34f3de

Thanks!

**Pull request closed without merging**